sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InMemoryColumnarQuerySuite.scala

```diff
@@ -479,4 +485,43 @@ class InMemoryColumnarQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-22673: InMemoryRelation should utilize existing stats of the plan to be cached") {
+    withSQLConf("spark.sql.cbo.enabled" -> "true") {
+      withTempDir { workDir =>
+        withTable("table1") {
+          val workDirPath = workDir.getAbsolutePath + "/table1"
```

Comment:
seems you can use `withTempPath`, which just gives you a path string without creating it.

Reply:
then we don't need to append `/table1` to the path